src/mmrelay/matrix_utils.py

```
@@ -2614,7 +3154,57 @@ async def matrix_relay(
+            if init_attempt < max_init_retries - 1:
+                logger.warning(
+                    f"Matrix client initialization failed (attempt {init_attempt + 1}/{max_init_retries}): {exc}, "
+                    f"retrying in {init_retry_delay}s..."
+                )
+                await asyncio.sleep(init_retry_delay)
+            else:
+                logger.exception(
+                    "Matrix client initialization failed after %s attempts: %s. "
+                    "Message to room %s may be lost.",
+                    max_init_retries,
+                    exc,
```

Ruff (0.14.14)

[warning] Redundant exception object included in `logging.exception` call

(TRY401)